Use `sync/atomic#Pointer` instead of our own wrapper

lippserd opened the pull request:
Go 1.19 introduced `sync/atomic#Pointer` among other things, so we no longer need to use the Atomic wrapper from our Icinga Go library.

Al2Klimov commented on pkg/icingadb/ha.go:
```diff
@@ -514,7 +515,7 @@ func (h *HA) signalHandover(reason string) {
 func (h *HA) signalTakeover(reason string) {
 	if !h.responsible {
-		h.state.Store(haState{
+		h.state.Store(&haState{
```
This is fine.

Al2Klimov commented on pkg/icingadb/ha.go:
```diff
@@ -121,7 +121,8 @@ func (h *HA) Takeover() chan string {
 func (h *HA) State() (responsibleTsMilli int64, responsible, otherResponsible bool) {
-	state, _ := h.state.Load()
+	state := h.state.Load()
+
```
This has been a zero value if uninitialized, now it's a nil ptr in this case. Now we have to make sure HA#State() don't get a nil ptr.

lippserd replied:
Done.

Al2Klimov commented on pkg/icingadb/ha.go:
```diff
@@ -428,7 +429,7 @@ func (h *HA) realize(
 		h.signalTakeover(takeover)
 	} else if otherResponsible {
-		if state, _ := h.state.Load(); !state.otherResponsible {
+		if state := h.state.Load(); !state.otherResponsible {
```
As we're now loading by-ref, `state.otherResponsible = true` below affects all readers. But this shouldn't be too bad as that's just a bool and we're storing it anyway right away.

lippserd replied:
Fixed.

yhabteab commented on pkg/icingadb/ha.go:
```diff
@@ -428,9 +433,17 @@ func (h *HA) realize(
 
 		h.signalTakeover(takeover)
 	} else if otherResponsible {
-		if state, _ := h.state.Load(); !state.otherResponsible {
+		var state haState
+		statePtr := h.state.Load()
+		if statePtr != nil {
+			// Dereference pointer to create a copy of the value it points to.
+			// Ensures that any modifications do not directly affect the original data unless explicitly stored back.
+			state = *statePtr
+		}
+
+		if !state.otherResponsible {
```
You don't change the state if you don't enter this `if` branch anyway, so I would handle this the same way as before.
```suggestion
		if state := h.state.Load(); state != nil && !state.otherResponsible {
			// Dereference pointer to create a copy of the value it points to.
			// Ensures that any modifications do not directly affect the original data unless explicitly stored back.
			newState := *state
			newState.otherResponsible = true
			h.state.Store(&newState)
		}
```

yhabteab replied:
Oh, never mind, I get it now why you did it that way 🙈

Al2Klimov reviewed (commented):
There's still

https://github.com/Icinga/icingadb/blob/34844764fc949e3b4b6607c700d052da3d9e7203/pkg/icingaredis/telemetry/heartbeat.go#L84

btw.

lippserd commented:
@Al2Klimov @yhabteab I also changed the type in `telemetry` and adjusted the implementation so that the `atomic.Pointer` usages initially hold the zero values of their types. In addition, copies of the pointer values are now only created if they are actually needed.

lippserd commented:
@yhabteab I'm sorry,  I had to push another change 😆.
